compiler/Makefile

```diff
@@ -115,12 +122,14 @@ $(CHPL_CONFIG_CHECK): | $(CHPL_BIN_DIR)
 	rm -f $(CHPL_CONFIG_CHECK_PREFIX)-*
 	echo $(CHPL_MAKE_COMPILER_SUBDIR) > $(CHPL_CONFIG_CHECK)
 
-$(CHPL): $(CHPL_OBJS) $(CHPL_CONFIG_CHECK) | $(CHPL_BIN_DIR)
+$(CHPL): TAGS $(CHPL_OBJS) $(CHPL_CONFIG_CHECK) | $(CHPL_BIN_DIR)
```

Comment:
No downside that I am aware of - but I made the swap and put it at the end, for the sake of "feeling right". 

I believe for parallel make, the left-to-right order is not ensured, without an explicit `.NOTPARALLEL` pseudotarget.